src/common/lib/util/utils.ts

```diff
@@ -174,6 +174,33 @@ export function arrSubtract<T>(arr1: Array<T>, arr2: Array<T>): Array<T> {
   return result;
 }
 
+export function arrIndexOfBy<T>(
+  arr: Array<T>,
+  iteratee: (value: T) => unknown,
+  elem: unknown,
+  fromIndex: number = 0,
+): number {
+  const len = arr.length;
+  for (; fromIndex < len; fromIndex++) {
+    if (iteratee(arr[fromIndex]) === elem) {
+      return fromIndex;
+    }
+  }
+  return -1;
+}
+
+export function arrUnique<T>(arr: Array<T>): Array<T> {
+  return arr.filter((value, index) => arr.indexOf(value) === index);
+}
+
+export function arrUniqueBy<T>(arr: Array<T>, iteratee: (value: T) => unknown): Array<T> {
+  return arr.filter((value, index) => arrIndexOfBy(arr, iteratee, iteratee(value)) === index);
+}
+
+export function arrIn(arr: Array<unknown>, val: unknown): boolean {
```

Comment:
I think you can safely use `includes` instead of `arrIn`

Reply:
yes, `arrIn` polyfill were removed in https://github.com/ably/ably-js/pull/1629/commits/e5961ead813a6435988679f22911ec232b41ef9e